File: packages/assets_util/android/src/main/java/net/yumnumm/assets_util/AssetsUtilPlugin.java
package net.yumnumm.assets_util;

import io.flutter.embedding.engine.plugins.FlutterPlugin;

/** Registers the Android library so Java classes ship in the app APK. */
public final class AssetsUtilPlugin implements FlutterPlugin {
  @Override
  public void onAttachedToEngine(FlutterPluginBinding binding) {
    // Context is provided by jni_flutter's JniFlutterPlugin.
  }

  @Override
  public void onDetachedFromEngine(FlutterPluginBinding binding) {}
}

File: packages/assets_util/android/src/main/kotlin/net/yumnumm/assets_util/AssetsUtilPlugin.kt
package net.yumnumm.assets_util

import io.flutter.embedding.engine.plugins.FlutterPlugin

/** Registers the Android library so Kotlin classes ship in the app APK. */
class AssetsUtilPlugin : FlutterPlugin {
  override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    // Context is provided by jni_flutter's JniFlutterPlugin.
  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
  }
}
